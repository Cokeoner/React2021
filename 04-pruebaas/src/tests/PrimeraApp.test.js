import React from 'react';
import { shallow } from 'enzyme'
// import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import PrimeraApp from  "../PrimeraApp";

describe('Pruebbas PrimeraApp', () => {
    
    // test('Debe de mostrar el mensaje "Hola soy goku"', () => {  
    //     const saludo = 'Hola soy goku';
    //     const {getByText} = render(<PrimeraApp saludo= { saludo }/>);
    //     expect( getByText(saludo)).toBeInTheDocument();
    // });

    test('Debe mostrar <PrimeraApp /> correctamente', () => {
       
        const saludo = 'Hola soy goku';
        const Wrapper = shallow( <PrimeraApp saludo= { saludo }/>)

        expect( Wrapper).toMatchSnapshot();

    });
    
    test('Dee de mostrar el susbstitulo enviado por props', () => {
        
        const saludo = 'Hola soy goku';
        const subtitulo = 'Soy un Subtitulo';
        const Wrapper = shallow( 
                <PrimeraApp 
                        saludo= { saludo }
                        subtitulo ={ subtitulo} 
                
                />)

        const TextoParrado = Wrapper.find('p').text();

        expect( TextoParrado).toBe(subtitulo);

    })
    
    

});