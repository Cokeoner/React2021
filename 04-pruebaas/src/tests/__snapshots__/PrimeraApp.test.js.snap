// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebbas PrimeraApp Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola soy goku
     
  </h1>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;
